dll/win32/shell32/dialogs/filetypes.cpp

```diff
@@ -1054,41 +1182,73 @@ EditTypeDlg_UpdateEntryIcon(HWND hwndDlg, PEDITTYPE_DIALOG pEditType,
 
 static BOOL
 EditTypeDlg_WriteClass(HWND hwndDlg, PEDITTYPE_DIALOG pEditType,
-                       LPCWSTR ClassKey, LPCWSTR ClassName, INT cchName)
+                       LPCWSTR TypeName, EDITTYPEFLAGS ETF)
 {
     PFILE_TYPE_ENTRY pEntry = pEditType->pEntry;
-
-    if (ClassKey[0] == 0)
-        return FALSE;
-
-    // create or open class key
+    LPCWSTR ClassKey;
+    HRESULT hr = GetClassKey(*pEntry, ClassKey);
+    BOOL OnlyExt = hr != S_OK;
     HKEY hClassKey;
-    if (RegCreateKeyExW(HKEY_CLASSES_ROOT, ClassKey, 0, NULL, 0, KEY_WRITE, NULL,
-                        &hClassKey, NULL) != ERROR_SUCCESS)
+    if (FAILED(hr) || RegCreateKeyExW(HKEY_CLASSES_ROOT, ClassKey, 0, NULL, 0,
+                                      KEY_QUERY_VALUE |KEY_WRITE, NULL,
+                                      &hClassKey, NULL) != ERROR_SUCCESS)
     {
         return FALSE;
     }
 
-    // create "DefaultIcon" key
-    if (pEntry->IconPath[0])
+    // Refresh the EditFlags
+    DWORD dw;
+    if (SHELL32_GetRegDW(hClassKey, L"EditFlags", dw, 0, FALSE) == S_OK)
+        pEntry->EditFlags = (dw & ~FTA_MODIFYMASK) | (pEntry->EditFlags & FTA_MODIFYMASK);
+
+    if (!OnlyExt)
+    {
+        // Set class properties
+        RegSetOrDelete(hClassKey, L"EditFlags", REG_DWORD, pEntry->EditFlags ? &pEntry->EditFlags : NULL, 4);
+        if (pEntry->IsExtension())
+        {
+            RegSetOrDelete(hClassKey, L"AlwaysShowExt", REG_SZ, (ETF & ETF_ALWAYSEXT) ? L"" : NULL, 0);
+        }
+        if (RegKeyExists(hClassKey, L"DocObject"))
+        {
+            HRESULT hr = SHELL32_GetRegDW(hClassKey, L"BrowserFlags", dw, 0, TRUE);
+            if (hr == S_OK || hr == HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND))
+            {
+                dw = (dw & ~8) | ((ETF & ETF_BROWSESAME) ? 0 : 8); // Note: 8 means NOT
```

Comment:
Aren't there human-readable defines for these flags? (8, and 4 used just below and above)

Reply:
Not for the 8, that value is undocumented except for useless KB articles that don't mention what the bits are called or what they actually do.